### PR TITLE
Result evaluation

### PR DESCRIPTION
**Pull request by princechinni**

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -15,19 +15,27 @@
...
-    "@azure/identity": "^4.2.0",
+    "@azure/identity": "^4.4.1",
     "@azure/storage-blob": "^12.23.0",
     "@sendgrid/mail": "^8.1.3",
     "axios": "^1.7.2",
     "azure-storage": "^2.10.7",
     "body-parser": "^1.20.2",
     "cors": "^2.8.5",
     "dotenv": "^16.4.5",
+<<<<<<< Updated upstream
```
Comment (anirudhcode7) [inline server/package.json]: These should not be appearing like this. Somewhere you didn't stash your changes properly and you pushed without resolving conflicts.

File: server/package.json [hunk 2]
```diff
@@ -15,19 +15,27 @@
...
     "express": "^4.21.0",
     "fluent-ffmpeg": "^2.1.3",
     "fs": "^0.0.1-security",
+=======
+    "express": "^4.18.2",
+    "express-session": "^1.18.0",
+    "fluent-ffmpeg": "^2.1.3",
+    "fs": "^0.0.1-security",
+    "google-auth-library": "^9.14.1",
+>>>>>>> Stashed changes
```
Comment (anirudhcode7) [inline server/package.json]: Identify common packages, and use the latest version. Remove this >>> and === lines.

File: server/services/azureService.js [hunk 3]
```diff
@@ -417,49 +457,32 @@ const combineAllChunksInToOneVideo = async (userId, jobId, questionId) => {
...
   );
-  const jobDir = path.join(__dirname, "tempChunks", userId, jobId);
+  console.log("Updated answer in DB successfully");
 
-  try {
-    fs.rmSync(questionDir, { recursive: true, force: true });
-  } catch (err) {
-    console.error(`Error deleting questionId directory ${questionDir}:`, err);
-  }
+  // await uploadFinalVideoToAzure(
```
Comment (anirudhcode7) [inline server/services/azureService.js]: Why is this commented?

File: server/services/openAIService.js [hunk 4]
```diff
@@ -147,85 +151,122 @@ const evaluateAnswer = async (interviewText) => {
...
+      interview_id: interviewId,
+      question_scores: [questionData],
+    });
+  } else {
+    result.question_scores.push(questionData);
+    await result.save();
+  }
+};
+
+const evaluateTranscriptionForAllQuestions = async (
```
Comment (anirudhcode7) [inline server/services/openAIService.js]: Change the name of the function to one question. As you are evaluating for only one question, not all questions.

File: server/services/openAIService.js [hunk 5]
```diff
@@ -147,85 +151,122 @@ const evaluateAnswer = async (interviewText) => {
...
+  const questionDoc = await Question.findById(questionId);
+
+  const question = questionDoc.question;
+
+  const response = await evaluateAnswer({ question, answer: transcription });
+
+  const scores = response.scores;
+  const feedback = response.feedback;
+
+  await updateOrCreateResult(interviewId, questionId, scores, feedback);
```
Comment (anirudhcode7) [inline server/services/openAIService.js]: Why not evaluate for each question and answer, and then update all of them together. Try reducing the number of DB operations.

**Review by anirudhcode7 — CHANGES_REQUESTED** [review]

File: server/package.json (hunk 2 above)
Comment (princechinni) [reply server/package.json]: After pushing the code I saw that and I have resolved it

File: server/services/azureService.js (hunk 3 above)
Comment (princechinni) [reply server/services/azureService.js]: While testing I commented it forgot to uncomment

File: server/services/openAIService.js (hunk 4 above)
Comment (princechinni) [reply server/services/openAIService.js]: Ok

File: server/services/openAIService.js (hunk 5 above)
Comment (princechinni) [reply server/services/openAIService.js]: I have used the existing code. If you want me to change I will change it.

**Pull request merged**